connect/src/routes/route.ts

```diff
@@ -1,45 +1,78 @@
-import { Network, ProtocolName } from "@wormhole-foundation/sdk-base";
-import { ChainAddress, Signer, TokenId, TransactionId } from "@wormhole-foundation/sdk-definitions";
+import { Network, Platform, ProtocolName } from "@wormhole-foundation/sdk-base";
+import {
+  ChainAddress,
+  ChainContext,
+  Signer,
+  TokenId,
+  TransactionId,
+} from "@wormhole-foundation/sdk-definitions";
 import { Wormhole } from "../wormhole";
-import { TransferReceipt } from "../wormholeTransfer";
-import { UnknownRoute } from "./resolver";
+import { TransferQuote, TransferReceipt } from "../wormholeTransfer";
 
 export interface RouteTransferRequest {
   from: ChainAddress;
   to: ChainAddress;
   amount: bigint;
   source: TokenId | "native";
-  destination?: TokenId;
+  destination?: TokenId | "native";
 }
 
-export type ValidationResult<E = Error> = { valid: true } | { valid: false; error: E };
+export type ValidationResult<T> =
+  | { quote: TransferQuote; options: T; valid: true }
+  | { options: T; valid: false; error: Error };
+
+export type RouteConstructor<N extends Network, OP> = {
+  new (
+    wh: Wormhole<N>,
+    request: RouteTransferRequest,
+    fromChain?: ChainContext<N>,
+    toChain?: ChainContext<N>,
+  ): Route<N, OP>;
+  // Check if this route is supported for the given transfer request
+  // e.g. check if the protocols on the specific chains are supported
+  isSupported(fromChain: ChainContext<N, Platform>, toChain: ChainContext<N, Platform>): boolean;
+  // Get the default options for this route, useful to prepopulate a form
+  getDefaultOptions(): OP;
```

Comment:
I think having this be a static method makes sense, but I disagree with what you did to `isSupported`